CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/resourcetypes/CVKCommandBuffer.java

```diff
@@ -146,8 +180,294 @@ public void FinishRecord(){
+            renderPassBeginInfo.renderPass(renderPass);
+            renderPassBeginInfo.pClearValues(clearValues);
+            renderPassBeginInfo.framebuffer(frameBuffer);
+
+            VkRect2D renderArea = renderPassBeginInfo.renderArea();
+            renderArea.offset().set(0, 0);
+            renderArea.extent().set(width, height);
+
+            vkCmdBeginRenderPass(vkCommandBuffer, renderPassBeginInfo, flags);
+
+            renderPassBeginInfo.free();
+	}
+	
+	public void endRenderPassCmd(){
+		
+	    vkCmdEndRenderPass(vkCommandBuffer);
+	}
+	
+	public void BindComputePipelineCmd(long pipeline){
+		
+	    vkCmdBindPipeline(vkCommandBuffer, VK_PIPELINE_BIND_POINT_COMPUTE, pipeline);
+	}
+	
+	public void BindGraphicsPipelineCmd(long pipeline){
+		
+	    vkCmdBindPipeline(vkCommandBuffer, VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline);
+	}
+	
+	public void viewPortCmd(int width, int height, MemoryStack stack){
+		
+	    VkViewport.Buffer viewport = VkViewport.callocStack(1, stack);
+            viewport.x(0.0f);
+            viewport.y(0.0f);
+            viewport.width(width);
+            viewport.height(height);
+            viewport.minDepth(0.0f);
+            viewport.maxDepth(1.0f);
+            
+            vkCmdSetViewport(vkCommandBuffer, 0, viewport);
+	}
+	
+	public void scissorCmd(VkExtent2D extent, MemoryStack stack){
+		
+	    VkRect2D.Buffer scissor = VkRect2D.callocStack(1, stack);
+            scissor.offset(VkOffset2D.callocStack(stack).set(0, 0));
+            scissor.extent(extent);
 
-    public static CVKCommandBuffer Create(final CVKDevice cvkDevice, final int level, final String debugName) {
+            vkCmdSetScissor(vkCommandBuffer, 0, scissor);
+	}
+	
+	public void pushConstantsCmd(long pipelineLayout, int stageFlags, ByteBuffer data){
+		
+		vkCmdPushConstants(vkCommandBuffer, // The buffer to push the matrix to
+				pipelineLayout,     // The pipeline layout
+				stageFlags,         // Flags
+				0,                  // Offset
+				data);              // Push constant data
+	}
+//	
+//	public void bindVertexInputCmd(long vertexBuffer, long indexBuffer){
+//		
+//		LongBuffer offsets = memAllocLong(1);
+//		offsets.put(0, 0L);
+//		LongBuffer pVertexBuffers = memAllocLong(1);
+//		pVertexBuffers.put(0, vertexBuffer);
+//		
+//		vkCmdBindVertexBuffers(vkCommandBuffer, 0, pVertexBuffers, offsets);
+//		vkCmdBindIndexBuffer(vkCommandBuffer, indexBuffer, 0, VK_INDEX_TYPE_UINT32);
+//		
+//		memFree(pVertexBuffers);
+//		memFree(offsets);
+//	}
+//	
+	public void bindVertexInputCmd(long vertexBuffer){
+		
+		LongBuffer offsets = memAllocLong(1);
+		offsets.put(0, 0L);
+		LongBuffer pVertexBuffers = memAllocLong(1);
+		pVertexBuffers.put(0, vertexBuffer);
+		
+		vkCmdBindVertexBuffers(vkCommandBuffer, 0, pVertexBuffers, offsets);
+		
+		memFree(pVertexBuffers);
+		memFree(offsets);
+	}
+	
+	public void bindComputeDescriptorSetsCmd(long pipelinyLayout, long[] descriptorSets){
```

Comment:
This one too.

BTW we don't use any compute shaders so why does this exist?

Reply:
Yep will do in another pass